src/components/LayoutTree.tsx

```diff
@@ -446,6 +452,55 @@ const TreeNode = ({
     return lastPatches?.some(patch => deleteActions.includes(patch.actions[0]))
   })
 
+  const [showLineEndFauxCaret, setShowLineEndFauxCaret] = useState(false)
+  const [showLineStartFauxCaret, setShowLineStartFauxCaret] = useState(false)
+
+  // Hide the faux caret when typing occurs.
+  editingValueStore.subscribe(() => {
+    if (isTouch && isSafari() && caretRef.current) {
+      caretRef.current.style.display = 'none'
+      setShowLineStartFauxCaret(false)
+      setShowLineEndFauxCaret(false)
+    }
+  })
+
+  // If the thought isCursor and edit mode is on, position the faux cursor at the point where the
+  // selection is created.
+  useEffect(() => {
+    if (isTouch && isSafari() && caretRef.current) {
```

Comment:
Rather than wrap the entire effect in a conditional, I prefer to invert the condition and short circuit at the top. This avoids the extra indentation level and makes it a bit easier to read.

```ts
if (!isTouch || !isSafari() || !caretRef.current) return
```